### PR TITLE
Uninstall /can fail/

### LINKED ISSUE #149666: flutter_engine_group_performance failing consistently on mokey
It was passing on MotoG4

https://ci.chromium.org/ui/p/flutter/builders/luci.flutter.staging/Linux_mokey%20flutter_engine_group_performance

### PR DESCRIPTION
**Pull request by jtmcdole**
Error != Exception.

Fixes #149666

### CODE REVIEW

**Review by reidbaker — APPROVED** [review]

File: dev/devicelab/bin/tasks/flutter_engine_group_performance.dart [hunk 1]
```diff
@@ -20,13 +20,7 @@ Future<void> _withApkInstall(
...
   final AndroidDevice device = await devices.workingDevice as AndroidDevice;
   await device.unlock();
-  try {
-    // Force proper cleanup before trying to install app. If uninstall fails,
-    // we log exception and proceed with running the test.
-    await device.adb(<String>['uninstall', bundleName]);
-  } on Exception catch (error) {
-    print('adb uninstall failed with exception: $error. Will proceed with test run.');
-  }
+  await device.adb(<String>['uninstall', bundleName], canFail: true);
```
Comment (zanderso) [inline dev/devicelab/bin/tasks/flutter_engine_group_performance.dart:23]: Lgtm w/ question

Does canFail:true still log the uninstall failure?

**Review by zanderso — APPROVED** [review]

File: dev/devicelab/bin/tasks/flutter_engine_group_performance.dart (hunk 1 above)
Comment (reidbaker) [reply dev/devicelab/bin/tasks/flutter_engine_group_performance.dart:23]: I had the same thought and decided this was still net better.

Comment (jtmcdole) [reply dev/devicelab/bin/tasks/flutter_engine_group_performance.dart:23]: No. _execute just returns the code. I can test for it and log if you want.

```dart
  if (exitCode != 0 && !canFail) {
    fail('Executable "$executable" failed with exit code $exitCode.');
  }
```

Comment (jtmcdole) [reply dev/devicelab/bin/tasks/flutter_engine_group_performance.dart:23]: ... is what I would say if adb returned the exit code, but it returns stdout.

adb uninstall in the LOGs will print an error:

> [2024-09-17 11:02:18.801185] [STDOUT] stdout: Failure [DELETE_FAILED_INTERNAL_ERROR]

**Pull request merged**